atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java

```diff
@@ -233,13 +234,14 @@ public Void apply(DSLContext ctx) {
         if (Iterables.isEmpty(rows)) {
             return ImmutableMap.of();
         }
-        return run(new Function<DSLContext, Map<Cell, Value>>() {
-            @Override
-            public Map<Cell, Value> apply(DSLContext ctx) {
+
+        Map<Cell, Value> toReturn = new HashMap<>();
+        for (List<byte[]> part : Iterables.partition(rows, PARTITION_SIZE)) {
```

Comment:
doesn't look that there is a way to batch on columns here since there is no straightforward way to limit size on the jdbc queries. I will keep the impl as is(without unnecessary batching) and recommend turning on sweep if someone faces issues. Also, will make the`PARTITION_SIZE` configurable.